example/lib/ios_popover/_ios_toolbar.dart

```diff
@@ -0,0 +1,594 @@
+///
+/// Buttons to access the next/previous page are included by default.
+///
+/// Either [pages] of [children] must be non-null.
+class _IosToolbarContent extends StatefulWidget {
+  const _IosToolbarContent({
+    Key? key,
+    this.pages,
+    this.children,
+  })  : assert(pages != null || children != null),
+        assert(pages == null || children == null),
+        super(key: key);
+
+  /// List of items to be auto-paginated.
+  final List<Widget>? children;
+
+  /// List of pages containing the menu items.
+  final List<MenuPage>? pages;
+
+  @override
+  State<_IosToolbarContent> createState() => _IosToolbarContentState();
+}
+
+class _IosToolbarContentState extends State<_IosToolbarContent> {
+  final _MenuPageController _controller = _MenuPageController();
+
+  /// Creates the button which points to the previous page.
+  Widget _buildPreviousPageButton() {
+    return TextButton(
+      style: TextButton.styleFrom(
+        padding: const EdgeInsets.all(0),
+        minimumSize: const Size(30, 0),
+      ),
+      onPressed: _controller.goToPrevious,
+      child: Icon(
+        Icons.arrow_left,
+        color: _controller.isFirstPage ? Colors.grey : Colors.white,
+      ),
+    );
+  }
+
+  /// Creates the button which points to the next page.
+  Widget _buildNextPageButton() {
+    return TextButton(
+      style: TextButton.styleFrom(
+        padding: const EdgeInsets.all(0),
+        minimumSize: const Size(30, 0),
+      ),
+      onPressed: _controller.goToNext,
+      child: Icon(
+        Icons.arrow_right,
+        color: _controller.isLastPage ? Colors.grey : Colors.white,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: _controller,
+      builder: (context, child) {
+        return _IosPaginatedMenu(
+          controller: _controller,
+          previousButton: _buildPreviousPageButton(),
+          nextButton: _buildNextPageButton(),
+          pages: widget.pages,
+          children: widget.children,
+        );
+      },
+    );
+  }
+}
+
+/// Controls a paginated menu.
+class _MenuPageController extends ChangeNotifier {
+  int _currentPage = 1;
+  int _maxPages = 1;
+
+  /// Indicates if the current page is the last page.
+  bool get isLastPage => _currentPage == _maxPages;
+
+  /// Indicates if the current page is the first page.
+  bool get isFirstPage => _currentPage == 1;
+
+  /// Current page in the menu.
+  int get currentPage => _currentPage;
+  set currentPage(int value) {
+    if (value != _currentPage) {
+      _currentPage = value;
+      notifyListeners();
+    }
+  }
+
+  /// Number of pages in the menu.
+  int get pageCount => _maxPages;
+  set pageCount(int value) {
+    if (value != _maxPages) {
+      _maxPages = value;
+      notifyListeners();
+    }
+  }
+
+  /// Advances to the next page, notifying listeners.
+  void goToNext() {
+    if (currentPage < pageCount) {
+      currentPage++;
```

Comment:
As a matter of style, I try not to use the postfix `++`. Instead, please use `currentPage += 1`, which has a clear order of operations.